File: 04 - ORIENTAÇÃO A OBJETOS/nodeJS - JavaScript_/Exercicio 4/Com Pacotes/Aluno.js
let Pessoa = require('./Pessoa')

module.exports = class Aluno extends Pessoa {
    constructor(nome, idade, matricula){
        super(nome, idade)
        this.matricula = matricula
    }

    dizerMatricula(){
        return 'Minha matricula é: '+this.matricula
    }

    dizerNomeMatricula(){
        return super.dizerNome()+' e '+this.dizerMatricula()
    }
}

File: 04 - ORIENTAÇÃO A OBJETOS/Java/Exercicio 4/Aluno.java
/*

 */
public class Aluno extends Pessoa {
    int matricula;

    //CONSTRUTOR DA CLASSE HERDADA E DA PRÓPRIA CLASSE
    public Aluno(String nome, int idade, int matricula) {
        super(nome, idade);
        this.matricula = matricula;
    }
    
    String dizerMatricula(){
        return "Minha matricula é: "+this.matricula;
    }
    
    void dizerNomeMatricula(){
        System.out.println (super.dizerNome() + ". "+this.dizerMatricula());
    }
}
